chapters/image_accceses.md

```diff
@@ -0,0 +1,97 @@
+# Image Accesses
+
+SPIR-V has various [Image Instructions](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_image_instructions) that will interact with `OpTypeImage`. This chapter aims to give some extra context around these instructions.
+
+> There is an [example.frag](./examples/image_accesses/example.frag) (and [example.spv](./examples/image_accesses/example.spv)) GLSL shader that has an example of various ways images can be accessed in SPIR-V.
+
+# Images as Handles
+
+An `OpTypeImage` can be thought of as a handle the underlying data. To help illustrate this, here is a small program that writes to an image
+
+```glsl
+layout(set = 0, binding = 1, rgba8) uniform writeonly image2D image;
+void main() {
+    imageStore(image, ivec2(0, 0), vec4(1.0));
+}
+```
+
+Taking a closer look at the `OpVariable` we see it is of type [UniformConstant](../chapters/storage_class.md#uniformconstant) which is suppose to be "read-only" (yet there is a clear `OpImageWrite` occuring here).
+
+This is possible because the handle to the image is read-only, but the underlying texel data of the image is mutable.
```

Comment:
I would appreciate clarification on this point. What would it mean for an image variable to have a mutable storage class (such as `Uniform`)? Is that legal?

I still feel some confusion about which modifiers apply to which data.

- `NonReadable` and `NonWritable` apply to the underlying texel data.
- Storage class applies to the metadata, but I got the idea from Alan that the fields in `OpTypeImage` are all immutable, regardless of storage class. If that is true, what does the storage class really say about the image variable?